ENG-1946 - feat: introduce React package documentation (#143)
Basis-Theory/docs: source/includes/elements/react/_package.md

# React Package

> NPM

```shell
npm install --save @basis-theory/basis-theory-react
```

> Yarn

```shell
yarn add @basis-theory/basis-theory-react
```

> **Basis Theory React** is an [open source](https://github.com/Basis-Theory/basis-theory-react) [package](https://www.npmjs.com/package/@basis-theory/basis-theory-react) designed to allow you to easily integrate Basis Theory SDK and Elements features in your React solution.

Reviewer: It's hard to discern that there are two links here, one for the public repo and another for the NPM package.

![image](https://user-images.githubusercontent.com/43392371/149369546-1fa771fa-9139-4ea8-9e1e-5e616f0a4f33.png)

Author: I agree. Maybe a change in the verbiage could solve it. Any suggestions?

> Under this section, you can find out how to initialize and access `BasisTheory` instance using <a href="https://reactjs.org/docs/hooks-intro.html" target="_blank">Hooks</a>, pass it through your component tree with <a href="https://reactjs.org/docs/context.html" target="_blank">Context</a> and declare Elements <a href="https://reactjs.org/docs/components-and-props.html" target="_blank">Component</a> in your code. 
> 
> Creating, mounting, updating and unmounting elements is as simple as declaring their tags and passing props.

Reviewer 2: ```suggestion
This section provides specific documentation for how to initialize a `BasisTheory` instance using <a href="https://reactjs.org/docs/hooks-intro.html" target="_blank">Hooks</a>, pass it to your component tree with <a href="https://reactjs.org/docs/context.html" target="_blank">Context</a> and finally declare Elements <a href="https://reactjs.org/docs/components-and-props.html" target="_blank">Component</a> in your code use the underlying Element. 

Creating, mounting, updating and unmounting Elements is as simple as declaring them as a typical React Components and passing props.
```

Author: Not sure about "finally". Two reasons: you are not required to use the Context Provider to declare Elements, so they are not consecutive actions; and can make one feel like the integration is long.

<aside class="notice">

>   <span>You don't have to install <code>@basis-theory/basis-theory-js</code> separately when using our React package, unless you need to direct import something from it.</span>

Reviewer 2: ```suggestion
  <span>When using our React package, you do not need to import <code>@basis-theory/basis-theory-js</code> separately, unless you need to direct import a capability from it.</span>
```

Author: I want to stick to "install" in the first sentence.

Reviewer 2: ✅
